front/lib/resources/data_source_view_resource.ts

```diff
@@ -32,6 +33,20 @@ import {
 import type { ResourceFindOptions } from "@app/lib/resources/types";
 import type { VaultResource } from "@app/lib/resources/vault_resource";
 
+export const getDataSourceCategory = (
```

Comment:
This code already exist here: https://github.com/dust-tt/dust/blob/f66d3788c68cca3d30d67d005f1e3d5ece9bd844/front/lib/api/vaults.ts#L6-L18